client/src/components/Pagination.tsx

```diff
@@ -72,6 +70,143 @@ export default function Pagination({
   );
 }
 
+interface PaginationNavProps {
+  activePage: number;
+  ariaLabel?: string;
+  innerClassName?: string;
+  itemsCountPerPage: number;
+  maxPages?: number;
+  onChange: (pageNumber: number) => void;
+  totalItemsCount: number;
+}
+
+const PaginationNav = memo(function PaginationNav({
+  activePage,
+  ariaLabel = "Page navigation",
+  innerClassName,
+  itemsCountPerPage,
+  onChange,
+  maxPages = 5,
+  totalItemsCount,
+}: PaginationNavProps) {
+  if (itemsCountPerPage === 0 || totalItemsCount === 0 || maxPages <= 0) {
+    return null;
+  }
+  const totalPages = Math.ceil(totalItemsCount / itemsCountPerPage);
+  if (totalPages <= 1) {
+    return null;
+  }
+
+  const half = Math.floor(maxPages / 2);
+  const start = Math.max(1, activePage - half);
+  const end = Math.min(totalPages, start + maxPages - 1);
+  const startPage =
+    end === totalPages ? Math.max(1, totalPages - maxPages + 1) : start;
+
+  const pages = [];
+
+  // First page button: only show if the first page is not already shown
+  if (startPage !== 1) {
+    pages.push(
+      <PaginationElement
+        ariaLabel="first page"
+        onClick={() => onChange(1)}
+        key="first"
+      >
+        ⟪
+      </PaginationElement>
+    );
+  }
+
+  // Previous page button: only show if there's a previous page
+  if (activePage > 1) {
+    pages.push(
+      <PaginationElement
+        ariaLabel="previous page"
+        onClick={() => onChange(activePage - 1)}
+        key="prev"
+      >
+        ⟨
+      </PaginationElement>
+    );
+  }
+
+  // Page number buttons
+  for (let pageNumber = startPage; pageNumber <= end; pageNumber++) {
+    pages.push(
+      <PaginationElement
+        ariaLabel={`page ${pageNumber.toString()} of ${totalPages}`}
+        onClick={() => onChange(pageNumber)}
+        className={pageNumber === activePage ? "active" : ""}
```

Comment:
This is where you use `cx`:
```suggestion
        className={cx(pageNumber === activePage && "active")}
```